packages/medusa/src/repositories/product-category.ts

```diff
@@ -1,5 +1,47 @@
-import { EntityRepository, TreeRepository } from "typeorm"
+import { EntityRepository, TreeRepository, Brackets, ILike } from "typeorm"
 import { ProductCategory } from "../models/product-category"
+import { ExtendedFindConfig, Selector } from "../types/common"
 
 @EntityRepository(ProductCategory)
-export class ProductCategoryRepository extends TreeRepository<ProductCategory> {}
+export class ProductCategoryRepository extends TreeRepository<ProductCategory> {
+  public async getFreeTextSearchResultsAndCount(
+    options: ExtendedFindConfig<ProductCategory, Selector<ProductCategory>> = {
+      where: {},
+    },
+    q: string | undefined
+  ): Promise<[ProductCategory[], number]> {
+    const options_ = { ...options }
+
+    const queryBuilder = this.createQueryBuilder("product_category")
```

Comment:
suggestiom(non blocking): put `product_category` into const alias. so that you can use it here and at the end for the relations. Also, it simplifies future update